packages/@aws-cdk/aws-ec2/test/test.vpc.ts

```diff
@@ -415,6 +415,18 @@ export = {
       }
       test.done();
     },
+
+    'natGateways = 0 requires there to be no PRIVATE subnets'(test: Test) {
+      const stack = getTestStack();
+      test.throws(() => {
+        new Vpc(stack, 'VPC', {
+          natGateways: 0,
+        });
+      }, /make sure you don't configure any PRIVATE subnets/);
```

Comment:
Perhaps provide the rationale for this in the error message or an accompanying link to an issue or doc